### PR TITLE
Write update.log to datadir

### LINKED ISSUE #317: updater/update.log has to be stored in the data folder
The updater/update.log file should NOT be publicly accessible as it leaks information and also breaks code signing. It needs to be stored in the data folder.

Ref https://github.com/owncloud/core/issues/23857#issuecomment-209312462

cc @karlitschek @jospoortvliet @VicDeo

### PR DESCRIPTION
**Pull request by VicDeo**
Fixes #317 

Init logger when datadir is available. Fallback to stderr.
All errors happened when no  logger exists goes to stdout.

### CODE REVIEW

File: src/Console/Application.php [hunk 1]
```diff
@@ -135,6 +166,23 @@ protected function doRunCommand(Command $command, InputInterface $input, OutputI
 	}
 
 	/**
+	 * @param string $baseDir
+	 */
+	protected function initLogger($baseDir){
```
Comment (DeepDiver1975) [inline src/Console/Application.php]: Does this affect the web updater as well?

Comment (VicDeo) [inline src/Console/Application.php]: @DeepDiver1975 yes. Web UI executes commands via AJAX requests 
https://github.com/owncloud/updater/blob/master/src/Controller/IndexController.php#L127

Comment (DeepDiver1975) [inline src/Console/Application.php]: thx

**Comment by DeepDiver1975** [comment]
:+1:

**Comment by VicDeo** [comment]
Stable9 https://github.com/owncloud/updater/pull/321

**Pull request merged**